.github/workflows/test.yml

```diff
@@ -0,0 +1,28 @@
+name: test
+on:
+  push:
+    branches:
+      - "**"
+  pull_request:
+    branches:
+      - develop
+      - master
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Go setup
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.18
+
+      - name: Build
+        run: go build -v ./...
+
+      - name: Test
+        run: go test -v -race ./...
```

Comment:
Maybe add `-failfast` ?

Reply:
Good idea, since we only care about whether all tests pass or not -- hunting down additional failing tests will be done locally anyway.